src/app/modules/payments/stripe.controller.ts

```diff
@@ -191,17 +193,76 @@ export const getPaymentReceipt: ControllerHandler<{
       logger.info({
         message: 'Received receipt url from Stripe webhook',
         meta: {
-          action: 'getPaymentReceipt',
+          action: 'checkPaymentReceiptStatus',
           receiptUrl,
         },
       })
-      res.status(StatusCodes.OK).send({ receipt: receiptUrl })
+
+      return res.status(StatusCodes.OK).json({ isReady: true })
     })
     .mapErr((error) => {
       logger.error({
         message: 'Error retrieving receipt URL',
         meta: {
-          action: 'getPaymentReceipt',
+          action: 'checkPaymentReceiptStatus',
+          formId: formId,
+          submissionId: submissionId,
+        },
+        error,
+      })
+      return res.status(StatusCodes.NOT_FOUND).json({ message: error })
+    })
+}
+
+export const downloadPaymentReceipt: ControllerHandler<{
+  formId: string
+  submissionId: string
+}> = (req, res) => {
+  const { formId, submissionId } = req.params
+  logger.info({
+    message: 'downloadPaymentReceipt endpoint called',
+    meta: {
+      action: 'downloadPaymentReceipt',
+      formId: formId,
+      submissionId: submissionId,
+    },
+  })
+
+  return StripeService.getReceiptURL(formId, submissionId)
+    .map((receiptUrl) => {
+      logger.info({
+        message: 'Received receipt url from Stripe webhook',
+        meta: {
+          action: 'downloadPaymentReceipt',
+          receiptUrl,
+        },
+      })
+      // retrieve receiptURL as html
+      return (
+        axios
+          .get<string>(receiptUrl)
+          // convert to pdf and return
+          .then((receiptUrlResponse) => {
+            const html = receiptUrlResponse.data
+            return tracer.trace('generatePdfFromHtml', () => {
+              const pdfBufferPromise = generatePdfFromHtml(html)
+              return pdfBufferPromise
+            })
```

Comment:
It's probably better to create the DD span in the function `generatePdfFromHtml()` itself, so it will create spans regardless of who the caller is.

Reply:
[e8982e6](https://github.com/opengovsg/FormSG/pull/5837/commits/e8982e6dabba323362c014aae8d95ab16fa3dd61)